java/tools/src/java/org/apache/orc/tools/json/JsonSchemaFinder.java

```diff
@@ -247,15 +249,41 @@ public void addFile(String filename) throws IOException {
     } else {
       reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
     }
-    addFile(reader);
+    addFile(reader, filename);
   }
 
-  public void addFile(java.io.Reader reader) throws IOException {
+  public void addFile(java.io.Reader reader, String filename) {
     JsonStreamParser parser = new JsonStreamParser(reader);
-    while (parser.hasNext()) {
-      records += 1;
-      mergedType = mergeType(mergedType, pickType(parser.next()));
+    try {
+      while (parser.hasNext()) {
+        mergedType = mergeType(mergedType, pickType(parser.next()));
+        records += 1;
+      }
+    } catch (JsonParseException e) {
+      printParseExceptionMsg(e, filename);
+    }
+  }
+
+  private void printParseExceptionMsg(JsonParseException e, String filename) {
+    System.err.printf(
+        "A JsonParseException was thrown while processing the %dth record of file %s.%n",
+        records + 1, filename);
+
+    String pattern = "at line (\\d+) column (\\d+)";
+    Pattern r = Pattern.compile(pattern);
+    Matcher m = r.matcher(e.getMessage());
+    int line;
+    int column;
+    if (m.find( )) {
+      line = Integer.parseInt(m.group(1));
+      column = Integer.parseInt(m.group(2));
+      if (line == 1 && column == 1) {
+        System.err.printf("File %s is empty.", filename);
```

Comment:
Please add a new line at the end.
```
-        System.err.printf("File %s is empty.", filename);
+        System.err.printf("File %s is empty.\n", filename);
```

Otherwise, it looks like the following.
```
$ java -jar tools/target/orc-tools-1.8.0-SNAPSHOT-uber.jar json-schema ~/empty.json
Reading file /Users/dongjoon/empty.json
A JsonParseException was thrown while processing the 1th record of file /Users/dongjoon/empty.json.
File /Users/dongjoon/empty.json is empty.%
```

Reply:
c4599399a0a1f812b0068b87bdf409ae4f021d95 I fixed it